drivers/platform/stm32/stm32_pwm.c

```diff
@@ -107,7 +107,7 @@ static int32_t stm32_init_timer(struct stm32_pwm_desc *desc,
 	if (sparam->get_timer_clock) {
 		timer_frequency_hz = sparam->get_timer_clock();
 		timer_frequency_hz *= sparam->clock_divider;
-		timer_frequency_hz /= NO_OS_BIT(sparam->prescaler);
+		timer_frequency_hz /= sparam->prescaler;
```

Comment:
Since you are at it, could you please also document here what this value should be ?
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/platform/stm32/stm32_pwm.h#L72

It's unclear for the user, so specifying the range in the comment i linked above would be great.

Reply:
I've mentioned a range here. Does it look ok?

Reply:
Looking again at the stm32 pwm implementation, i think it's over complicated for the user.

Yes, it's configurable, the user can specify the stm32_pwm_init_param `prescaler` and `clock_divider`, but i think they should be computed by the stm32_pwm implementation based on the info in no_os_pwm_init_param.
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/platform/stm32/stm32_pwm.c#L108C3-L110

This is probably a topic for another PR.

Reply:
I felt the same too. Few points on this
1)  We could probably have an implementation where the prescaler is calculated based on the PWM period.
2) The clock_divider parameter we set from the top layer does not get reflected into the HAL layer. It is being hardcoded [here](https://github.com/analogdevicesinc/no-OS/blob/master/drivers/platform/stm32/stm32_pwm.c#L157). So should we really expose that parameter via the [stm32_pwm.h](https://github.com/analogdevicesinc/no-OS/blob/master/drivers/platform/stm32/stm32_pwm.h#L85)?

Reply:
1. agreed
2. no need to expose it, agreed

I will merge this PR now and save this conclusion in a backlog. Would be great if you'd implement it but really it's up to you and your priorities.

Reply:
thanks!